server/gitrest/packages/gitrest-base/src/utils/helpers.ts

```diff
@@ -254,12 +254,35 @@ export async function checkSoftDeleted(
     const lumberjackProperties = {
         ...getLumberjackBasePropertiesFromRepoManagerParams(repoManagerParams),
     };
+    const metric = Lumberjack.newLumberMetric(
```

Comment:
We might not "complete" the metric In case an error happens in lines 260-262. To avoid that, we could wrap the main logic in a try-catch, and mark the metric as error in the catch block, followed by a rethrow to make sure we propagate the error up (which I guess is the current behavior)

Reply:
Good catch, thanks!